### ISSUE TITLE
LOOT will not open. No fixes found work.

### ISSUE THREAD

**Issue by KingofQuietDarkness** [loot/loot#1045]
Hello. I have used loot for a good long time now, and had been planning to use it for even longer. however, yesterday, after going here to download the installer again, after accidentally uninstalling it, it no longer works. it pulls up the run program as admin window, but loot itself will never open as a program. 

I have deleted the app data in local files in my user files, I have uninstalled and reinstalled at least seven or eight times now. i have run as administrator and as regular on and off all day yesterday and narly all day today until now.

Nothing works. nothing fixes my problem. Upon opening Vortex, I actually et an error message, which is thus: 

Error: libloadorder failed to create a game handle. Details: stream did not contain valid UTF-8: Libloadorder error

Upon installing loot, it tells me it is an error code 740, which is just elevating the process to fix it.

Nothing works. i am very angry and extremely irritated. it worked fine Thursday, used it at least seven times that day as i was altering things to make my games work significantly. and now i can no longer make this work at all. 

I very much would like an answer and some help please. it would be very helpful to my sanity.

**Comment by CptMcSplody** [comment]
@KingofQuietDarkness Can you turn on debug logging in LOOT and upload a copy of the debug log?

> Upon installing loot, it tells me it is an error code 740, which is just elevating the process to fix it.
This would appear to be an issue with windows, have you tried running the LOOT installer as admin?

**Comment by KingofQuietDarkness** [comment]
I'm going to be honest, i'm not sure how to turn on the debug logging feature. But yes, I have tried running the installer as an admin.

**Comment by CptMcSplody** [comment]
![image](https://user-images.githubusercontent.com/1944639/48304777-58b3ff80-e517-11e8-8a21-e172a99e0496.png)
In settings, but it's not really an issue with the LOOT program. It's UAC in windows http://windows-exe-errors.com/windows-error-740-the-requested-operation-requires-elevation/

**Comment by CptMcSplody** [comment]
The location of the generated debug log can be opened here 
![image](https://user-images.githubusercontent.com/1944639/48304802-d1b35700-e517-11e8-9cd3-7fc6877d0e49.png)

**Comment by KingofQuietDarkness** [comment]
Ahhhh, I see. Well, i do not intend to sound rude, but I have run it as administrator in every conceivable way. The loot program itself will not open. It just opens the yes or no options for running it as an admin. I'm sorry, should have been more clear.

**Comment by CptMcSplody** [comment]
@KingofQuietDarkness test with this [snapshot](https://bintray.com/loot/snapshots/download_file?file_path=loot_0.13.4-27-g4cd2631_update-api.7z) build and see if it runs.

**Comment by KingofQuietDarkness** [comment]
Well, I tried to open it, and at first it pulled up the window asking me if i wanted to run it because it thought it was a virus, I hit run anyway, and then it still wouldn't open, tried running it as admin, now it's opened a window asking if it ran correclty or if i want to launch a troubleshooter. I'm using the troubleshooter, and i'm going to see what it tells me.

**Comment by KingofQuietDarkness** [comment]
Troubleshooter did not help, when I hit test the program it still would not open.

**Comment by pStyl3** [comment]
If you haven't already did so, please restart Windows and see if that helped.

**Comment by KingofQuietDarkness** [comment]
So,I restarted windows, and it still won't open. It works fine on my laptop, though, as i tested it earlier, with on errors.

**Comment by KingofQuietDarkness** [comment]
[LOOTDebugLog.txt](https://github.com/loot/loot/files/2568810/LOOTDebugLog.txt)
Here we go, I found my debug file finally.

**Comment by Ortham** [comment]
It doesn't look like that log was generated with debug logging enabled, there should be more in it. Interesting that you're getting the same error in LOOT as in Vortex though, looks like I was wrong about my hunch earlier. Please also upload your loadorder.txt for Skyrim, if you've got one.

**Comment by KingofQuietDarkness** [comment]
I'm not sure where that would be, sorry.

**Comment by KingofQuietDarkness** [comment]
Wait, no ,i founds it
[loadorder.txt](https://github.com/loot/loot/files/2568825/loadorder.txt)

**Comment by KingofQuietDarkness** [comment]
[loadorder.txt](https://github.com/loot/loot/files/2568826/loadorder.txt)
[loadorder.txt](https://github.com/loot/loot/files/2568828/loadorder.txt)

Oblivon and fallout new vegas as well, just in case.

**Comment by Ortham** [comment]
Well, they all look fine. I'll have to do some digging. If you can get a full debug log that would help.

**Comment by KingofQuietDarkness** [comment]
Well, that's slightly annoying, i'm going to be honest. Just my creating settings.toml, it works again. i'm very sorry for all the trouble. Thank you very much for your help, it's been worth it. also, sorry for my... let's say dense view of things.

**Comment by Ortham** [comment]
Did you create an empty settings.toml? If not, what did you put in it?

**Comment by KingofQuietDarkness** [comment]
I created a file named settings.toml, and that's all I did, clicked loot, and it launched up no problem.

**Comment by CptMcSplody** [comment]
So the `settings.toml` was not being created? @Ortham Could a windows permission issue or an anti-virus program prevent LOOT from creating the `settings.toml`?

**Comment by KingofQuietDarkness** [comment]
[LOOTDebugLog.txt](https://github.com/loot/loot/files/2570242/LOOTDebugLog.txt)
I ported my loot files from my laptop to my desktop to see if it would help. While it did not cause it work again, ((Still annoyed about my fix working exactly one day.)) it DID let me edit the settings.toml file finally to get a full ((I hope)) debug log, which is attached.

**Comment by Ortham** [comment]
I think there are two issues here, one is the error

```
[21:10:53.718213] [error]: libloadorder failed to create a game handle. Details: stream did not contain valid UTF-8
```

and the other is that LOOT doesn't start up sometimes, which sounds like #930. I think it was just coincidence that it worked after creating the `settings.toml`. Try running LOOT with the ``--single-process`` and/or ``--disable-gpu`` parameters.

**Comment by TanninOne** [comment]
Vortex itself writes loadorder.txt utf-8 encoded. Also, Vortex doesn't use libloadorder directly at all, it just uses it through the loot-api, which I don't build myself so it's using the version used in that.

**Comment by KingofQuietDarkness** [comment]
C:\Users\username\AppData\Local\LOOT, if that's what you mean for my  path? I also assume that if setting the paths for my games does not work, and I have to use the parameters you listed, that those are command line arguments? therefore, i need to use the cmd?

**Comment by Ortham** [comment]
Does your username contain any non-ASCII characters?

You can create a shortcut to LOOT.exe and add the parameters to the end of the Target field in the shortcut's properties.

**Comment by KingofQuietDarkness** [comment]
No, my username folder is just my first and last name, with a space between them.

After creating the shortcut, and trying both disable gpu and single process, it still will not work.

However, my debug log has reverted back to single line of text again.
i also noticed in the shortcut that loot is represented as C:\LOOT\LOOT.exe

Is that normal? If not, could that be my problem? I tried redownloading it again last night, and it told me after I hit finish that loot could not run due to an error revolving around loot having itself set like that, so.

**Comment by Ortham** [comment]
> i also noticed in the shortcut that loot is represented as C:\LOOT\LOOT.exe
> 
> Is that normal? If not, could that be my problem? I tried redownloading it again last night, and it told me after I hit finish that loot could not run due to an error revolving around loot having itself set like that, so.

I don't know where you've installed LOOT, I can't tell you if that's the right path or not, and I don't know the error you're referring to.

**Comment by KingofQuietDarkness** [comment]
Um. It's just in my c drive. I click Acer (C:), and everything except my debug log ad such is in there., in a folder named loot. My apologies if that was not very descriptive, I've never really been on a forum before, so I don't really know how to explain things here or anything. I mean no disrespect or anything like that. I also am not a programer. I just use the loot application to fix my load order.

**Comment by Ortham** [comment]
I think I'm missing something here. If you've installed LOOT to `C:\LOOT\`, then the shortcut pointing to `C:\LOOT\LOOT.exe` is fine. If you haven't installed it there, then it's not. `C:\LOOT\` isn't the usual location, if you run the LOOT installer it will put it in Program Files by default, so I assume you've put it in `C:\LOOT\` yourself.

As for the libloadorder error you're getting, I can only see it happening when comparing your `loadorder.txt` and `plugins.txt` for consistency, but the `loadorder.txt` you uploaded is valid UTF-8. Can you also upload your `plugins.txt` for Skyrim, in case the problem is with it?

**Comment by KingofQuietDarkness** [comment]
[plugins.txt](https://github.com/loot/loot/files/2578284/plugins.txt)
Here you go.

**Comment by Ortham** [comment]
I can't replicate your issue using that plugins.txt and loadorder.txt. Try removing `Ak’Chazar Follower.esp` from your Skyrim's Data folder, just to see if that's it. Can you give me a link to where that plugin is from? It doesn't come up in search results...

**Comment by KingofQuietDarkness** [comment]
https://josiencatraso.tumblr.com/post/146767922667/at-long-last-akchazar-a-telvanni-lich-custom

Here's a link to the mod itself, I'll remove it and see if it works. I'm very sorry for all the trouble.

**Comment by KingofQuietDarkness** [comment]
I believe you are right, because it works now, even if it is a little slow loading.

**Comment by KingofQuietDarkness** [comment]
I'll let you know if it keeps working.

**Comment by Ortham** [comment]
Well, I still can't replicate your issue even with that plugin installed and using your loadorder.txt and plugins.txt... There must be something else going on.

**Comment by Ortham** [comment]
OK, I spotted an inconsistency in the file encoding handling - when loading the load order, libloadorder will fall back to Windows-1252 if `loadorder.txt` isn't in UTF-8, but when creating the libloadorder game handle, there's a consistency check that doesn't include that fallback behaviour.

The `loadorder.txt` you uploaded is valid UTF-8, but all I can think of is that something else (not Vortex, thanks to Tannin's clarification) is writing a non-UTF-8 file but by the time you get the file to upload it, that's been replaced with a UTF-8 version.

Try putting `Ak’Chazar Follower.esp` back into your Data folder, and run [this](https://bintray.com/loot/snapshots/download_file?file_path=loot_0.13.4-29-g523c48e_update-api.7z) LOOT build, replacing its `loot_api.dll` with [this one](https://bintray.com/loot/snapshots/download_file?file_path=loot_api-0.13.8-54-gf304af2_update-libloadorder-win32.7z).

**Comment by KingofQuietDarkness** [comment]
So, just to give some updates on this, and where i am at now: I removed ak'chazar like you had asked, and it seemed to fix it the first time, and it's been running great yesterday and today. 

REINSTALLING Ak'Chazar, and using your build files still lets loot come up on my desktop pc, however, loot tells me this:

An error occurred while parsing the metadata list(s): yaml-cpp: error at line 1342, column 9: bad conversion: invalid condition syntax: Failed to parse condition "version("Unofficial Oblivion Patch.esp","3.4.0",>=)". Details: An error was encountered in the parser "SeparatedList" while parsing the expression "version("Unofficial Oblivion Patch.esp","3.4.0",>=)".

Try updating your masterlist to resolve the error. If the error is with your user metadata, this probably happened because an update to LOOT changed its metadata syntax support. Your user metadata will have to be updated manually.

To do so, use the 'Open Debug Log Location' in LOOT's main menu to open its data folder, then open your 'userlist.yaml' file in the relevant game folder. You can then edit the metadata it contains with reference to the documentation, which is accessible through LOOT's main menu.

You can also seek support on LOOT's forum thread, which is linked to on  LOOT's website.

So, i'm going to try and update my masterlist and see what happens.

**Comment by KingofQuietDarkness** [comment]
This error is on skyrim's page in loot

An error occurred while parsing the metadata list(s): yaml-cpp: error at line 627, column 9: bad conversion: invalid condition syntax: Failed to parse condition "version("Unofficial Skyrim Patch.esp", "2.0", >=)". Details: An error was encountered in the parser "SeparatedList" while parsing the expression "version("Unofficial Skyrim Patch.esp", "2.0", >=)".

Try updating your masterlist to resolve the error. If the error is with your user metadata, this probably happened because an update to LOOT changed its metadata syntax support. Your user metadata will have to be updated manually.

To do so, use the 'Open Debug Log Location' in LOOT's main menu to open its data folder, then open your 'userlist.yaml' file in the relevant game folder. You can then edit the metadata it contains with reference to the documentation, which is accessible through LOOT's main menu.

You can also seek support on LOOT's forum thread, which is linked to on LOOT's website.

**Comment by KingofQuietDarkness** [comment]
[LOOTDebugLog.txt](https://github.com/loot/loot/files/2582386/LOOTDebugLog.txt)

**Comment by Ortham** [comment]
OK, it looks like the new condition parser is more strict than the old one, I'll look into that, but the fact that LOOT even gets that far means the libloadorder fix has worked.

**Comment by Ortham** [comment]
Try replacing your `loot_api.dll` with [this one](https://bintray.com/loot/snapshots/download_file?file_path=loot_api-0.13.8-55-gdde77f6_update-libloadorder-win32.7z), it should fix the condition parsing errors.

**Comment by Ortham** [comment]
@KingofQuietDarkness can you confirm if the last build I linked to fixes the condition parsing errors?

**Comment by KingofQuietDarkness** [comment]
Yes ,it does. It works fine now. no problems so far. I would have answered sooner, but retail is a bitch.

**Comment by Ortham** [comment]
Great, I'm glad that's sorted. I'll keep this issue until the changes have been merged into `dev`.

**Comment by Ortham** [comment]
Fixed in ecbe050ad699fbde5ab13f7614bd80feea2ee182.
